test/FunctionalTests/Balancer/ConnectionTests.cs

```diff
@@ -47,6 +49,48 @@ namespace Grpc.AspNetCore.FunctionalTests.Balancer
     public class ConnectionTests : FunctionalTestBase
     {
 #if NET5_0_OR_GREATER
+        [Test]
+        public async Task Active_UnaryCall_ConnectTimeout_ErrorThrownWhenTimeoutExceeded()
+        {
+            // Ignore errors
+            SetExpectedErrorsFilter(writeContext =>
+            {
+                return true;
+            });
+
+            var tcs = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
+            string? host = null;
+            async Task<HelloReply> UnaryMethod(HelloRequest request, ServerCallContext context)
+            {
+                var protocol = context.GetHttpContext().Request.Protocol;
+
+                Logger.LogInformation("Received protocol: " + protocol);
+
+                await tcs.Task;
+                host = context.Host;
+                return new HelloReply { Message = request.Name };
+            }
+
+            // Arrange
+            using var endpoint = BalancerHelpers.CreateGrpcEndpoint<HelloRequest, HelloReply>(50051, UnaryMethod, nameof(UnaryMethod));
+
+            var handler = new SocketsHttpHandler()
+            {
+                // ConnectTimeout is so small that CT will always be canceled before Socket is used.
+                ConnectTimeout = TimeSpan.FromTicks(1),
```

Comment:
Doesn't TimeSpan.Zero work? We all know that using anything with real time will be flaky.
And Tick(1) actually means 15 milliseconds on windows which isn't super fast.

Reply:
TimeSpan.Zero doesn't work. The server has just started up so it should take longer than 15ms to establish the first connection. If it proves flaky, I'll come up with a better idea.